lemmatized_text/models.py

```diff
@@ -198,6 +207,50 @@ def update_token(self, user, token_index, node_id, resolved):
             resolved=resolved,
         )
 
+    def handle_edited_data(self, title, edits):
+        self.title = title
+
+        cleaned_edits = edits.replace("<p>", "").replace("</p>", "<br/>").replace("<br/>", "\n")
+        edit_parser = EditedTextHtmlParser(
+            token_node_dict=self.token_node_dict(),
+            lang=self.lang
+        )
+        edit_parser.feed(cleaned_edits)
+
+        # Trimming junk tokens that get appended to the end of the list
+        for token in reversed(edit_parser.lemmatized_text_data):
+            if token["word"] != "":
+                break
+            edit_parser.lemmatized_text_data.remove(token)
+        self.data = edit_parser.lemmatized_text_data
+
+        strip_parser = TagStripper()
+        strip_parser.feed(cleaned_edits)
+        self.original_text = strip_parser.get_data()
+        self.save()
+
+    def token_node_dict(self):
+        lemma_dict = defaultdict(list)
+        for token in self.data:
+            lemma_dict[token["word"]].append(token["node"])
+        return dict(lemma_dict)
+
+    def transform_data_to_html(self):
+        return "".join([
+            (
+                f"<span {self.format_token_key_value_pairs(token)}>"
+                f"{token['word']}</span><span follower='true'>"
+                f"{self.parse_following(token['following'])}</span>"
+            )
+            for token in self.data
+        ])
+
+    def parse_following(self, follower):
+        return follower.replace("\n", "<br/>")
+
+    def format_token_key_value_pairs(self, token):
+        return " ".join([f"{k}='{v}'" for k, v in token.items() if k not in ("word", "following")])
```

Comment:
Since `parse_following` and `format_token_key_value_pairs` appear to be pure helper functions for `transform_data_to_html`, you could consider nesting them inside instead of calling them as methods on the `LemmatizedText` model or extracting them as module-level functions. The only advantage is that it would make the dependency more explicit, and since the tests are only exercising `transform_data_to_html`, you can still be confident it's working as expected. It's also fine as-is, just thinking about how these are used compared to the other methods.

Reply:
Good feedback, thanks